api/src/pcapi/core/offers/api.py

```diff
@@ -1296,3 +1298,33 @@ def delete_price_category(offer: models.Offer, price_category: models.PriceCateg
     validation.check_price_categories_deletable(offer)
     db.session.delete(price_category)
     db.session.commit()
+
+
+def list_nearest_venues_for_offer(
+    offer: models.Offer,
+    latitude: float | None = None,
+    longitude: float | None = None,
+) -> list[offerers_models.Venue]:
+    if latitude is not None and longitude is not None:
+        center = func.ST_MakePoint(longitude, latitude)
+    else:
+        center = func.ST_MakePoint(offer.venue.longitude, offer.venue.latitude)
+    venues = (
+        offerers_models.Venue.query.join(offerers_models.Venue.offers)
+        .join(offerers_models.Venue.managingOfferer)
+        .options(contains_eager(offerers_models.Venue.managingOfferer))
+        .filter(models.Offer.id != offer.id)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏J'ai du mal à conprendre ce `filter`
c'est le `offer_id` de la `Venue` ? du `Offerer` ?

Reply:
c'est l'id de l'offre "source" dont on se sert pour trouver d'autres lieux. c'est juste pour exclure du resulat le lieu qui propose l'offre "source"